src/ui/backend/BaseBacking.js

```diff
@@ -62,36 +83,39 @@ export default class BaseBacking {
 
   copy () {
     var copy = {};
+    for (var key in BASE_STYLE_PROPS) {
+      copy[key] = this[key];
+    }
 
-    for (var i = 0; i < styleKeyList.length; i++) {
-      var key = styleKeyList[i];
+    for (var i = 0; i < extendedStylePropList.length; i++) {
+      var key = extendedStylePropList[i];
       copy[key] = this[key];
     }
 
     return copy;
   }
 
   update (style) {
-    for (var i = 0; i < styleKeyList.length; i++) {
-      var key = styleKeyList[i];
+    for (var key in style) {
+      if (this[key] !== undefined) {
```

Comment:
Worth noting that using `void 0` here would be faster than `undefined` since it's a technically a global variable. Our compiler used to do this automatically, so we should look at that sometime. `undefined` is a little more straightforward to read.